mail/util.py

```diff
@@ -0,0 +1,62 @@
+# taken (and _very_ slightly adjusted)
+# from https://docs.aws.amazon.com/ses/latest/dg/smtp-credentials.html
+
+import base64
+import hashlib
+import hmac
+
+import ctx
+
+SMTP_REGIONS = [
+    'us-east-2',       # US East (Ohio)
+    'us-east-1',       # US East (N. Virginia)
+    'us-west-2',       # US West (Oregon)
+    'ap-south-1',      # Asia Pacific (Mumbai)
+    'ap-northeast-2',  # Asia Pacific (Seoul)
+    'ap-southeast-1',  # Asia Pacific (Singapore)
+    'ap-southeast-2',  # Asia Pacific (Sydney)
+    'ap-northeast-1',  # Asia Pacific (Tokyo)
+    'ca-central-1',    # Canada (Central)
+    'eu-central-1',    # Europe (Frankfurt)
+    'eu-west-1',       # Europe (Ireland)
+    'eu-west-2',       # Europe (London)
+    'sa-east-1',       # South America (Sao Paulo)
+    'us-gov-west-1',   # AWS GovCloud (US)
+]
+
+# These values are required to calculate the signature. Do not change them.
+DATE = "11111111"
+SERVICE = "ses"
+MESSAGE = "SendRawEmail"
+TERMINAL = "aws4_request"
+VERSION = 0x04
+
+
+def _sign(key, msg):
+    return hmac.new(key, msg.encode('utf-8'), hashlib.sha256).digest()
+
+
+def calculate_ses_key(secret_access_key: str, region:str) -> str:
+    if region not in SMTP_REGIONS:
+        raise ValueError(f"The {region} Region doesn't have an SMTP endpoint.")
+
+    signature = _sign(("AWS4" + secret_access_key).encode('utf-8'), DATE)
+    signature = _sign(signature, region)
+    signature = _sign(signature, SERVICE)
+    signature = _sign(signature, TERMINAL)
+    signature = _sign(signature, MESSAGE)
+    signature_and_version = bytes([VERSION]) + signature
+    smtp_password = base64.b64encode(signature_and_version)
+    return smtp_password.decode('utf-8')
+
+
+def ses_key_for_aws_config(aws_config_name: str, cfg_factory=None) -> str:
+    if not cfg_factory:
+        cfg_factory = ctx.cfg_factory()
+
+    aws_config = cfg_factory.aws(aws_config_name)
+
+    return calculate_ses_key(
```

Comment:
inline. although, thinking of it, I think this module is not a good place to add this code anyhow.

how about adding this to `model/email.py` instead? this code is not really related to smtp/email-sending, but rather to AWS/SES (which might also disqualify model/email.py as a good fit; still, I think it is more adequate than mail-package